pkg/apiserver/rest/dualwriter_test.go

```diff
@@ -86,6 +87,12 @@ func TestCompare(t *testing.T) {
 			input2:   exampleObjGen2,
 			expected: true,
 		},
+		{
+			name:     "should return true when objects are the same, but different TypeMeta (kind and apiversion)",
```

Comment:
If anything except spec is different, Compare will return true right? It might be worth reflecting that in the name of the test and make those fields different in the two objects being compared. So change not just kind and apiversion but also status and object meta (anything else?).